classes/local/api/tracker.php

```diff
@@ -45,7 +45,52 @@ class tracker
      * @return void As the insertion is done through the {js} template API.
      */
 
-    private static function generate_auth_token($api_base_url, $payload)
+    public static function fetchSecureToken($external_session_id, $external_attendee_id)
+    {
+        $api_base_url = trim(get_config('quizaccess_proctor', 'proview_callback_url'));
+        $auth_payload = new \stdClass();
+        $auth_payload->username = trim(get_config('quizaccess_proctor', 'proview_admin_username'));
+        $auth_payload->password = trim(get_config('quizaccess_proctor', 'proview_admin_password'));
+        $auth_response = self::generate_auth_token($api_base_url, $auth_payload);
+        $auth_token = json_decode($auth_response)->access_token;
+        $proctor_token = trim(get_config('local_proview', 'token'));
+        $url = $api_base_url . '/token/playback';
+
+        $data = array(
+            'proctor_token' => $proctor_token,
+            'validity' => 120,
+            'external_session_id' => $external_session_id,
+            'external_attendee_id' => $external_attendee_id
+        );
+        $options = array(
+            'http' => array(
+                'method' => 'POST',
+                'header' => "Content-Type: application/json\r\n" .
+                    "Authorization: Bearer " . $auth_token,
+                'content' => json_encode($data)
+            )
+        );
+        $context = stream_context_create($options);
+        $response = file_get_contents($url, false, $context);
```

Comment:
@rohansharmasitoula What is the benefit of using `stream_context_create` and `file_get_contents` instead of curl here?

Reply:
Using file_get_contents seemed to be easier than Curl, was unaware about the pros of curl. Changed to curl implementation.